src/test/ui/lazy_normalization_consts/trait-resolution-breakage.rs

```diff
@@ -0,0 +1,18 @@
+// check-pass
+
+trait Trait<T> {
+    const ASSOC_CONST: usize = 0;
+}
+
+impl Trait<()> for u8 {}
+
+// `u8::ASSOC_CONST` is resolved today, but will be ambiguous
+// under lazy normalization.
```

Comment:
It *ought* to be ambiguous today, I'd say...

Reply:
Yeah, but it compiles on stable :cry: iirc you noted somewhere that this kind of local reasoning should get linted against/slowly phased out. Not completely sure where or when though